sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/MethodParameterUnused.vb

```diff
@@ -0,0 +1,185 @@
+            Console.WriteLine("foo")
+		End Sub
+
+        Public Function DoSomething6(ByVal a As Integer) As Integer
+            Return 1
+		End Function
+
+        Protected Sub DoSomething7(ByVal a As Integer)
+            Console.WriteLine("foo")
+		End Sub
+
+        Protected Function DoSomething8(ByVal a As Integer) As Integer
+            Return 1
+		End Function
+
+        Friend Sub DoSomething9(ByVal a As Integer)
+            Console.WriteLine("foo")
+		End Sub
+
+        Friend Function DoSomething10(ByVal a As Integer) As Integer
+            Return 1
+		End Function
+
+        <Obsolete()>
+        Private Sub Attribute1(ByVal a As Integer) ' Compliant because of the attribute
+            Console.WriteLine("foo")
+		End Sub
+
+        <Obsolete()>
+        Private Function Attribute2(ByVal a As Integer) As Integer ' Compliant because of the attribute
+            Return 1
+		End Function
+
+        Private Sub Empty1(ByVal a As Integer) ' Compliant because there is no statements
+		End Sub
+
+        Private Sub OnlyThrows1(ByVal a As Integer) ' Compliant because it only throws NotImplementedException
+            Throw New NotImplementedException()
+		End Sub
+
+        Private Function OnlyThrows2(ByVal a As Integer) As Integer ' Compliant because it only throws NotImplementedException
+            Throw New NotImplementedException()
+		End Function
+
+        Private Sub MyApplication_Startup(ByVal sender As Object, ByVal e As EventArgs) ' Compliant because it is event handler
+            Console.WriteLine()
+        End Sub
+
+        Private Class PrivateClass
+            Public Overridable Sub Overridable1(ByVal a As Integer) ' Compliant because it is Overridable
+                Console.WriteLine("foo")
+		    End Sub
+
+            Public Overridable Function Overridable2(ByVal a As Integer) As Integer ' Compliant because it is Overridable
+                Return 1
+		    End Function
+        End Class
+    End Class
+
+    Structure ValidCodeStructure
+        Private Sub DoSomething1(ByVal a As Integer)
+            Dim x = a
+		End Sub
+
+        Private Function DoSomething2(ByVal a As Integer) As Integer
+            Return a
+		End Function
+    End Structure
+
+    Module ValidCodeModule
+        Private Sub DoSomething1(ByVal a As Integer)
+            Dim x = a
+		End Sub
+
+        Private Function DoSomething2(ByVal a As Integer) As Integer
+            Return a
+		End Function
+    End Module
+
+    MustInherit Class AbstractValidCode
+        MustOverride Sub MyAbstractMethod(ByVal a As Integer)
+    End Class
+
+    Public Interface IPerson
+        Function GetSomething(ByVal age As Integer) As Integer
+    End Interface
+
+    Public Class Person
+        Implements IPerson
+
+        Public Function GetSomething(age As Integer) As Integer Implements IPerson.GetSomething ' Compliant because it's an interface implementation
+            Return 42
+        End Function
```

Comment:
add another method inside this class that doesn't Implement anything and is Noncompliant